### DISCUSSION TITLE
[DONE] Support for non-Earth CRS?

### DISCUSSION THREAD

**Discussion by jlaura** [developmentseed/titiler#375] [Feature Request]
I am testing TiTiler with an S3 hosted COG and am running into some (not unexpected) issues with the fact that the data are of another body (Europa in this case. Also tested Mars).

I am seeing the following error when attempting to hit the /cog/bounds endpoint.

    File "rasterio/_err.pyx", line 215, in rasterio._err.exc_wrap_pointer
    rasterio._err.CPLE_NotSupportedError: Cannot find coordinate operations from 'PROJCRS["Equirectangular EUROPA",BASEGEOGCRS["GCS_EUROPA",DATUM["D_EUROPA",ELLIPSOID["EUROPA_localRadius",1560800,0,LENGTHUNIT["metre",1,ID["EPSG",9001]]]],PRIMEM["Reference_Meridian",0,ANGLEUNIT["degree",0.0174532925199433,ID["EPSG",9122]]]],CONVERSION["unnamed",METHOD["Equidistant Cylindrical (Spherical)",ID["EPSG",1029]],PARAMETER["Latitude of 1st standard parallel",0,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8823]],PARAMETER["Longitude of natural origin",180,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8802]],PARAMETER["False easting",0,LENGTHUNIT["metre",1],ID["EPSG",8806]],PARAMETER["False northing",0,LENGTHUNIT["metre",1],ID["EPSG",8807]]],CS[Cartesian,2],AXIS["easting",east,ORDER[1],LENGTHUNIT["metre",1,ID["EPSG",9001]]],AXIS["northing",north,ORDER[2],LENGTHUNIT["metre",1,ID["EPSG",9001]]]]' to 'EPSG:4326'

I am surprised, because I get a valid result when using a marblecutter instance (hosted by a third-party provider; I would rather self host I think).

I am honestly not even sure which project to start working with to try and address this issue. RasterIO? rio_tiler? titiler? Any thoughts appreciated!

Also, here is the direct link to the data if anyone is interested! https://asc-jupiter.s3-us-west-2.amazonaws.com/europa/galileo_voyager/controlled_mosaics/11ESCOLORS01-02_GalileoSSI_Equi-cog.tif

**Comment by vincentsarago** [comment]
@jlaura this is really interesting. 
The problem comes from rio-tiler https://github.com/cogeotiff/rio-tiler/blob/master/rio_tiler/io/cogeo.py#L103-L105 where we transform the projection to WGS84. I think there is a quick fix to this as we did in morecantile https://github.com/developmentseed/morecantile/blob/master/morecantile/models.py#L141-L155

thanks for sharing the dataset, I'll see what I can do with it.

Next step. to visualize the data on a map you'll need a custom TMS for europa or mars. I wonder if you have already created it?

**Reply by jlaura** [reply]
@vincentsarago Thanks for the info! I'll go ahead and open an issue over on rio_tiler to see about getting a fix put in. 

When you ask about a TMS, are you referring to: https://developmentseed.org/titiler/examples/code/tiler_with_custom_tms/ and the need to create a custom CRS? Or something else?

**Comment by vincentsarago** [comment]
Note: this was continued over https://github.com/cogeotiff/rio-tiler/issues/428

with rio-tiler v3.0.0, non earth CRS should be supported 🥳

**Comment by vincentsarago** [comment]
done in 0.4.0
